### PR TITLE
[improve] #21580 @Deprecated added for newAck() methods

### LINKED ISSUE #21580: Clean up method used to create ack command
### Search before asking

- [X] I searched in the [issues](https://github.com/apache/pulsar/issues) and found nothing similar.

### Motivation

Now we have 8 methods to create an ack command. We should consider to check which one is redundant.
image
See more details at https://github.com/apache/pulsar/pull/21419.

### Solution

Only keep `public static ByteBuf newMultiMessageAck(long consumerId, TxnID txnID,
            List<Triple<Long, Long, ConcurrentBitSetRecyclable>> entries, long requestID) ` and mark other methods as deprecated.

### Alternatives

_No response_

### Anything else?

_No response_

### Are you willing to submit a PR?

- [X] I'm willing to submit a PR!

### PR DESCRIPTION
**Pull request by ubo-dev**
Fixes #21580 

Main Issue: #21580 

### Motivation

Now we have 8 methods to create an ack command. We should consider to check which one is redundant.

### Modifications

@Deprecated annotation added for required methods.

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

### Does this pull request potentially affect one of the following parts:

*If the box was checked, please highlight the changes*

- [ ] Dependencies (add or upgrade a dependency)
- [ ] The public API
- [ ] The schema
- [ ] The default values of configurations
- [ ] The threading model
- [ ] The binary protocol
- [ ] The REST endpoints
- [ ] The admin CLI options
- [ ] The metrics
- [ ] Anything that affects deployment

### Documentation

- [ ] `doc` 
- [ ] `doc-required` 
- [x] `doc-not-needed` 
- [ ] `doc-complete` 

### Matching PR in forked repository

PR in forked repository: https://github.com/ubo-dev/pulsar/pull/2

### CODE REVIEW

File: pulsar-common/src/main/java/org/apache/pulsar/common/protocol/Commands.java [hunk 1]
```diff
@@ -976,6 +976,7 @@ public static ByteBuf newLookupErrorResponse(ServerError error, String errorMsg,
         return serializeWithSize(newLookupErrorResponseCommand(error, errorMsg, requestId));
     }
 
+    @Deprecated
```
Comment (tisonkun) [inline pulsar-common/src/main/java/org/apache/pulsar/common/protocol/Commands.java:979]: With reason or alternative suggestions?

Ditto others.

Comment (ubo-dev) [reply pulsar-common/src/main/java/org/apache/pulsar/common/protocol/Commands.java:979]: Thanks!

**Review by RobertIndie — COMMENTED** [review]
It's better to bring this to the mailing list to gain more visibility.

**Comment by ubo-dev** [comment]
> It's better to bring this to the mailing list to gain more visibility.

@RobertIndie Hi, I'm pretty new here, should I try to do that, how is the process?

**Comment by ubo-dev** [comment]
@tisonkun I can handle the comments is there any templates that I should follow and should I open new PR for comments can I merge it with this one somehow?

**Comment by ubo-dev** [comment]
@tisonkun 
`/**
     * @deprecated use {@link #newMultiMessageAck(long consumerId, TxnID txnID, List<Triple<Long, Long, ConcurrentBitSetRecyclable>> entries, long requestID)
     */`
is there anything I should add?

**Pull request closed without merging**